src/gc.h

```diff
@@ -257,6 +258,18 @@ typedef struct {
     pagetable1_t *meta1[REGION2_PG_COUNT];
 } pagetable_t;
 
+typedef struct {
+    _Atomic(size_t) bytes_mapped;
+    _Atomic(size_t) bytes_resident;
+    _Atomic(size_t) bytes_freed;
+    _Atomic(size_t) bytes_allocd;
+    _Atomic(size_t) bytes_mallocd;
+    _Atomic(size_t) malloc_bytes_freed;
+    _Atomic(size_t) pages_perm_allocd;
+    _Atomic(size_t) heap_size;
+    _Atomic(size_t) heap_target;
+} gc_heapstatus_t;
```

Comment:
Since we only care about `gc_heap_stats.heap_size` atomically aggregated across all threads, whereas everything else is already tracked in ptls, would it be better to reduce this to just that one parameter?

Reply:
I think I will just keep `mapped` `resident` and target/size.